agent/acs/session/payload_responder.go

```diff
@@ -140,7 +140,7 @@ func (pmHandler *payloadMessageHandler) addPayloadTasks(payload *ecsacs.PayloadM
 
 		// Add ENI information to the task struct.
 		for _, acsENI := range task.ElasticNetworkInterfaces {
-			eni, err := ni.ENIFromACS(acsENI)
+			eni, err := ni.InterfaceFromACS(acsENI)
```

Comment:
Just a note that since `ni` in shared ecs-agent module is already consumed by Fargate we will need to implement this change in Fargate once the changes in this PR is merged and consumed.

Reply:
Please be sure to test against Fargate and submit any required changes to that codebase as you merge anything `/ecs-agent` (aka shared lib) related.  Both EC2 and Fargate tests should pass after this change is shipped.

Reply:
Yes, there will be separate PRs/CRs which takes care of the agent integrations. I will make sure the changes incorporate the testing plan which was previously reviewed by the team.